src/common/pool_map.c

```diff
@@ -1416,6 +1417,15 @@ add_domains_to_pool_buf(struct pool_map *map, struct pool_buf *map_buf,
 		map_comp.co_fseq = 1;
 		map_comp.co_nr = node.fdn_val.dom->fd_children_nr;
 
+		if (map != NULL) {
+			already_in_map = pool_map_find_domain(map,
+							      PO_COMP_TP_RACK,
+							      map_comp.co_id,
+							      NULL);
+			if (already_in_map > 0)
+				map_comp.co_status = PO_COMP_ST_UPIN;
```

Comment:
why not use the domain status in the pool map? or I miss sth?

Reply:
Further up (lines 1379-1386) we end up deciding the status based on whether there's already a map. I was making assumptions about the two states above -- that it needs to be one or the other. Copying the existing state sounds more correct, I can change it.